stdlib/public/core/UnsafePointer.swift.gyb

```diff
@@ -420,12 +420,15 @@ public struct ${Self}<Pointee>
 
   /// Return `end - self`.
   public func distance(to x: ${Self}) -> Int {
-    return x - self
+    return Int(Builtin.sub_Word(Builtin.ptrtoint_Word(x._rawValue),
+        Builtin.ptrtoint_Word(_rawValue)))
```

Comment:
I agree. I have a note to rewrite that builtin call and a few others; just trying not change too many layers at once here (the diff you're seeing above is just stripping away a call to the operator that I removed).